apps/web/src/components/Basenames/UsernameProfileSectionHeatmap/index.tsx

```diff
@@ -162,25 +162,53 @@ export default function UsernameProfileSectionHeatmap() {
     };
   };
 
+  type EtherscanEnvelope = {
+    status?: '1' | '0';
+    message?: string;
+    result?: unknown;
+    data?: { result?: unknown };
+  };
+
+  const isObject = (value: unknown): value is Record<string, unknown> => {
+    return typeof value === 'object' && value !== null;
+  };
+
+  const hasDataEnvelope = (value: unknown): value is { data: EtherscanEnvelope } => {
+    return isObject(value) && 'data' in value && isObject((value as { data?: unknown }).data);
+  };
+
   const fetchTransactions = useCallback(
     async (apiUrl: string, retryCount = 3): Promise<Transaction[]> => {
       try {
         const response = await fetch(apiUrl);
-        const json = (await response.json()) as {
-          data: { result: Transaction[]; status: '1' | '0'; message: string };
-        };
+        const json: unknown = await response.json();
```

Comment:
It would be better to cast this to an object based on the shape etherscan returns